pkg/plugins/runtime/k8s/controllers/namespace_controller.go

```diff
@@ -63,6 +78,22 @@ func (r *NamespaceReconciler) Reconcile(req kube_ctrl.Request) (kube_ctrl.Result
 	}
 }
 
+func (r *NamespaceReconciler) hasNetworkAttachmentDefinition() (bool, error) {
+	crds := v1beta1.CustomResourceDefinitionList{}
+	err := r.Client.List(context.Background(), &crds)
+	if err != nil {
+		return false, errors.Wrap(err, "could not fetch CustomResourceDefinitionList")
+	}
+
+	for _, crd := range crds.Items {
+		if crd.GetName() == "network-attachment-definitions.k8s.cni.cncf.io" {
+			return true, err
+		}
+	}
```

Comment:
isn't it better to just try to `Get` this one resource instead of listing all CRDs?